opnode/l2/input_derivation_test.go

```diff
@@ -241,65 +125,71 @@ func TestUnmarshalLogEvent(t *testing.T) {
 	}
 }
 
-/*
-type DeriveUserDepositsTestCase struct {
-	name     string
-	input    *deriveTxsTestInput
-	expected []*types.Transaction
-}
-
-func TestDeriveUserDeposits(t *testing.T) {
-	testCases := []DeriveUserDepositsTestCase{
-		{"no deposits", GenerateTest(0, 0), []*types.Transaction{}},
-		{"success deposit", GenerateTest(1, 0), []*types.Transaction{nil}}, // TODO
-		{"failed deposit", GenerateTest(0, 1), []*types.Transaction{}},
-		{"many deposits", nil, nil}, // TODO
```

Comment:
Are these test cases being ran in `reading_test.go` or is that just testing round trip?

Reply:
See new `TestDeriveUserDeposits` below, the diff isn't clear here. Tested a bunch of edge cases with deposits success/fail/logs/non-deposit-logs/etc. And no need to generate a full block or hardcode transactions, since we first generate the deposits, then convert those to logs (along with other non-deposit logs etc. as the test case specifies), and then check if `DeriveUserDeposits` derives equal deposits from those logs.

It's indeed similar to the L1-info deposit round trip test, but for user-deposits here, instead of the L1-info deposit.

Reply:
Regarding the diff, #128 includes the changes and may be easier to review (also includes the `reading_test.go` etc.)